lib/Util/SearchEndpointTrait.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace MovingImage\Client\VMPro\Util;
+
+use MovingImage\Client\VMPro\Entity\VideosRequestParameters;
+use MovingImage\Client\VMPro\Exception;
+
+/**
+ * Helper methods for dealing with search endpoint.
+ */
+trait SearchEndpointTrait
+{
+    /**
+     * Creates an elastic search query from the provided array od parameters.
+     *
+     * @param array  $params
+     * @param string $operator
+     *
+     * @return string
+     */
+    private function createElasticSearchQuery(array $params, $operator = 'AND')
+    {
+        $filteredParams = [];
+        foreach ($params as $name => $value) {
+            if (empty($name) || empty($value)) {
+                continue;
+            }
+
+            $filteredParams[] = sprintf('%s:%s', $name, $value);
+        }
+
+        return sprintf(implode(sprintf(' %s ', $operator), $filteredParams));
```

Comment:
sprintf is extremely slower than normal concat, my suggestion if not to use it by default.
(Actually it's just good when you want to format a number or you get the string from somewhere else, like DB)